packages/pwa/app/utils/utils.js

```diff
@@ -158,7 +160,112 @@ export const capitalize = (text) => {
+}
+
+/**
+ * This function returns the url config from the current configuration
+ * @return {object} - url config
+ */
+export const getUrlConfig = () => {
+    const {app} = getConfig()
+    if (!app.url) {
+        throw new Error("Can't find any valid url config. Please check your configuration file.")
+    }
+    return app.url
+}
+
+/**
+ * This function return the regex for matching site and locale
+ * @param config
+ * @return {{searchMatcherForSite: RegExp, searchMatcherForLocale: RegExp, pathMatcher: RegExp}}
+ */
+export const getConfigMatcher = (config) => {
+    if (!config) {
+        throw new Error('Config is not defined.')
+    }
+
+    const allSites = getSites()
+    const siteIds = []
+    const siteAliases = []
+    const localesIds = []
+    const localeAliases = []
+    allSites.forEach((site) => {
+        siteAliases.push(site.alias)
+        siteIds.push(site.id)
+        const {l10n} = site
+        l10n.supportedLocales.forEach((locale) => {
+            localesIds.push(locale.id)
+            localeAliases.push(locale.alias)
+        })
+    })
+    const sites = [...siteIds, ...siteAliases].filter(Boolean)
+    const locales = [...localesIds, ...localeAliases].filter(Boolean)
+
+    // prettier-ignore
+    // eslint-disable-next-line
+    const searchPatternForSite = `site=(?<site>${sites.join('|')})`
+    // prettier-ignore
+    // eslint-disable-next-line
+    const pathPattern = `(?:\/(?<site>${sites.join('|')}))?(?:\/(?<locale>${locales.join("|")}))?(?!\\w)`
+    // prettier-ignore
+    // eslint-disable-next-line
+    const searchPatternForLocale = `locale=(?<locale>${locales.join('|')})`
+    const pathMatcher = new RegExp(pathPattern)
+    const searchMatcherForSite = new RegExp(searchPatternForSite)
+    const searchMatcherForLocale = new RegExp(searchPatternForLocale)
+    return {
+        pathMatcher,
+        searchMatcherForSite,
+        searchMatcherForLocale
+    }
+}
+
+/**
+ * Return the target locale object from a site supported locales
+ * @param site - site to look for the locale
+ * @param localeRef - the locale ref to look for in site supported locales
+ * @return {object|undefined}
+ */
+export const getLocaleFromSite = (site, localeRef) => {
+    if (!site) {
+        throw new Error('site is not defined')
+    }
+    return site.l10n.supportedLocales.find(
+        (locale) => locale.id === localeRef || locale.alias === localeRef
+    )
+}
+
+/**
+ * Return all possible locale ref of a site based on o ref.
+ * A ref could either be id or alias,
+ * we want to know all of them if available
+ * @param site
+ * @param localeRef
+ * @return {array} - list of possible refs of a locale (id and alias)
+ */
+export const getLocaleRefsFromSite = (site, localeRef) => {
+    const locale = getLocaleFromSite(site, localeRef)
+    if (!locale) {
+        throw new Error("Can't find any locale base on that localeRef")
+    }
+    return [locale?.id, locale?.alias].filter(Boolean)
+}
```

Comment:
I'm a little confused by this function. Why would be need to provide a locale reference if the function returns all the locale references for a site?

(side note, there is a typo in the jsdoc description of this function)

Also, `getLocaleFromSite` might be better names `getLocaleByReference` Where the jsdoc description might be something like "Given a site and a locale reference, return the locale object"

Reply:
My intention for this function is to avoid doing this every time we want to get a list of locale refs of a certain site.
```
const defaultLocale = getLocaleByReference(site, site.l10n.defaultLocale)
const defaultLocaleRefs = [defaultLocale.alias, defaultLocale.id].filter(Boolean)
```

However,  we only use it once for `buildPathWithUrlConfig` for now. I don't find it useful much, so I've removed it.

I've changed the function `getLocaleFromSite`  to `getLocaleByReference` as suggested.